packages/contracts-ops/contracts/scripts/UpgradeCallBatches.s.sol

```diff
@@ -0,0 +1,236 @@
+        governanceRouterImpl = governanceRouterUpgrade(domain).implementation;
+        // All replicas have the same implmentation
+        for (uint256 i; i < networksArray.length; i++) {
+            if (!compareStrings(networksArray[i], domain)) {
+                replicaImpl = replicaOfUpgrade(domain, networksArray[i])
+                    .implementation;
+                break;
+            }
+            homeImpl = homeUpgrade(domain).implementation;
+        }
+    }
+
+    function loadController(string memory domain) internal {
+        beaconController = address(getUpgradeBeaconController(domain));
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                       GOVERNANCE CALL GENERATORS
+    //////////////////////////////////////////////////////////////*/
+
+    function generateGovernanceCalls(string memory domain) internal {
+        title("BeaconController upgrade encoded calls for", domain);
+        console2.log(
+            "Domain Number: ",
+            vm.toString(uint256(getDomainNumber(domain)))
+        );
+        console2.log("Function signature: upgrade(address, address)");
+        console2.log(
+            "Arguments: <contract_beacon>, <new_implementation_address>"
+        );
+
+        upgradeHome = abi.encodeWithSignature(
+            "upgrade(address,address)",
+            homeBeacon,
+            homeImpl
+        );
+        console2.log("Upgrade Home");
+        console2.logBytes(upgradeHome);
+
+        upgradeReplica = abi.encodeWithSignature(
+            "upgrade(address,address)",
+            replicaBeacon,
+            replicaImpl
+        );
+        console2.log("Upgrade Replica");
+        console2.logBytes(upgradeReplica);
+        upgradeGovRouter = abi.encodeWithSignature(
+            "upgrade(address,address)",
+            governanceRouterBeacon,
+            governanceRouterImpl
+        );
+        console2.log("Upgrade Governance Router");
+        console2.logBytes(upgradeGovRouter);
+
+        upgradeBridgeRouter = abi.encodeWithSignature(
+            "upgrade(address,address)",
+            bridgeRouterBeacon,
+            bridgeRouterImpl
+        );
+        console2.log("Upgrade Bridge Router");
+        console2.logBytes(upgradeBridgeRouter);
+
+        upgradeTokenRegistry = abi.encodeWithSignature(
+            "upgrade(address,address)",
+            tokenRegistryBeacon,
+            tokenRegistryImpl
+        );
+        console2.log("Upgrade Token Registry");
+        console2.logBytes(upgradeTokenRegistry);
+
+        upgradeBridgeToken = abi.encodeWithSignature(
+            "upgrade(address,address)",
+            bridgeTokenBeacon,
+            bridgeTokenImpl
+        );
+        console2.log("Upgrade Bridge Token");
+        console2.logBytes(upgradeBridgeToken);
+        if (compareStrings(domain, localDomainName)) {
+            pushLocal(beaconController, upgradeBridgeToken);
```

Comment:
tbh this is a refactor i was thinking would make sense for CallBatch - just `pushCall` which decides whether the call is local or remote

Reply:
the local/remote in callbatch is really referring to governing/non-governing in most cases 🤔